packages/agents/src/tests/index.test.ts

```diff
@@ -0,0 +1,672 @@
+import { env } from "cloudflare:test";
+import { expect, describe, it, vi } from "vitest";
+import {
+  getAgentByName,
+  routeAgentRequest,
+  getCurrentAgent,
+  unstable_callable,
+  Agent,
+  StreamingResponse,
+  type Connection
+} from "../index.ts";
+import { camelCaseToKebabCase } from "../client.ts";
+import type { Env } from "./worker";
+
+declare module "cloudflare:test" {
+  interface ProvidedEnv extends Env {}
+}
+
+describe("Agent Core Functionality", () => {
+  describe("getAgentByName", () => {
+    it("should return an agent with specified name", async () => {
+      const agentName = "test-agent";
+      const agent = await getAgentByName(env.TEST_AGENT, agentName);
+
+      expect(agent).toBeDefined();
+      expect(agent).toHaveProperty("fetch");
+      expect(typeof agent.fetch).toBe("function");
```

Comment:
define a method on the agent that returns.name, and test it from the outside here